modules/core/02-client/client/cli/tx.go

```diff
@@ -461,3 +457,96 @@ func NewCmdSubmitUpgradeProposal() *cobra.Command {
+			`Example Upgraded Client State JSON: 
+{
+"@type":"/ibc.lightclients.tendermint.v1.ClientState",
+ "chain_id":"testchain1",
+"unbonding_period":"1814400s",
+"latest_height":{"revision_number":"0","revision_height":"2"},
+"proof_specs":[{"leaf_spec":{"hash":"SHA256","prehash_key":"NO_HASH","prehash_value":"SHA256","length":"VAR_PROTO","prefix":"AA=="},"inner_spec":{"child_order":[0,1],"child_size":33,"min_prefix_length":4,"max_prefix_length":12,"empty_child":null,"hash":"SHA256"},"max_depth":0,"min_depth":0},{"leaf_spec":{"hash":"SHA256","prehash_key":"NO_HASH","prehash_value":"SHA256","length":"VAR_PROTO","prefix":"AA=="},"inner_spec":{"child_order":[0,1],"child_size":32,"min_prefix_length":1,"max_prefix_length":1,"empty_child":null,"hash":"SHA256"},"max_depth":0,"min_depth":0}],
+"upgrade_path":["upgrade","upgradedIBCState"],
+}
+		`,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, err := client.GetClientTxContext(cmd)
+			if err != nil {
+				return err
+			}
+
+			cdc := codec.NewProtoCodec(clientCtx.InterfaceRegistry)
+
+			name := args[0]
+
+			height, err := strconv.ParseInt(args[1], 10, 64)
+			if err != nil {
+				return err
+			}
+
+			plan := upgradetypes.Plan{
+				Name:   name,
+				Height: height,
+			}
+
+			// attempt to unmarshal client state argument
+			var clientState exported.ClientState
+			clientContentOrFileName := args[2]
+			if err := cdc.UnmarshalInterfaceJSON([]byte(clientContentOrFileName), &clientState); err != nil {
+
+				// check for file path if JSON input is not provided
+				contents, err := os.ReadFile(clientContentOrFileName)
+				if err != nil {
+					return fmt.Errorf("neither JSON input nor path to .json file for client state were provided: %w", err)
+				}
+
+				if err := cdc.UnmarshalInterfaceJSON(contents, &clientState); err != nil {
+					return fmt.Errorf("error unmarshalling client state file: %w", err)
+				}
+			}
+
+			authority, _ := cmd.Flags().GetString(FlagAuthority)
+			if authority != "" {
+				if _, err = sdk.AccAddressFromBech32(authority); err != nil {
+					return fmt.Errorf("invalid authority address: %w", err)
+				}
+			} else {
+				authority = sdk.AccAddress(address.Module(govtypes.ModuleName)).String()
+			}
+
+			msg, err := types.NewMsgIBCSoftwareUpgrade(authority, plan, clientState)
+			if err != nil {
+				return err
+			}
+
+			if err = msg.ValidateBasic(); err != nil {
+				return err
+			}
+
+			proposal, err := govcli.ReadGovPropFlags(clientCtx, cmd.Flags())
+			if err != nil {
+				return err
+			}
+
+			if err := proposal.SetMsgs([]sdk.Msg{msg}); err != nil {
+				return fmt.Errorf("failed to create proposal message for scheduling an IBC software upgrade: %w", err)
+			}
+
+			return tx.GenerateOrBroadcastTxCLI(clientCtx, cmd.Flags(), proposal)
+		},
+	}
+
+	cmd.Flags().String(FlagAuthority, "", "The address of the client module authority (defaults to gov)")
+
+	flags.AddTxFlagsToCmd(cmd)
+	govcli.AddGovPropFlagsToCmd(cmd)
```

Comment:
needing to mark title as required here too yea?

Reply:
yea, there was some confusion on this, [see above conversation](https://github.com/cosmos/ibc-go/pull/4558#discussion_r1316947982), but now clear and I can add em back :)